src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -59,12 +59,17 @@
      * @param snapshotProvider data provider for the snapshots that are relevant for the node
      * @param tipsViewModel container that gets updated with the latest tips (transactions with no children)
      * @param transactionRequester used to request missing transactions from neighbors
+     * @param testnet <tt>true</tt> if we are in testnet mode, this caps {@code mwm} to {@value #TESTNET_MWM_CAP}
+     *                regardless of parameter input.
+     * @param mwm minimum weight magnitude: the minimal number of 9s that ought to appear at the end of the transaction
+     *            hash
      */
-    TransactionValidator(Tangle tangle, SnapshotProvider snapshotProvider, TipsViewModel tipsViewModel, TransactionRequester transactionRequester) {
+    TransactionValidator(Tangle tangle, SnapshotProvider snapshotProvider, TipsViewModel tipsViewModel, TransactionRequester transactionRequester, boolean testnet, int mwm) {
```

Comment:
I will follow your suggestion and change it as it is more consistent with the rest of the code and splitting the config objects reduces the possible hidden dependencies.

My three iotas: I prefer to make the used config parameters explicit. That is why:

* too many method arguments typically point out code that needs to change.
* testing is easier as you need only the direct objects and don't have to mock some containing class.
* the code is easier to read/understand as you explicitly see what objects are needed inside (otherwise you have to look inside the implementation).

Reply:
done.

Reply:
yes